---
title: "DataTable"
author: "Awais Choudhry"
date: "2017-5-24"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:


```{r, warning=FALSE, message=FALSE}
library(readxl)
library(magrittr)
library(ggplot2)
library(tibble)
library(JASPAR2016)
library(TFBSTools)

## loading in the counts of Sequences
library(readxl)
 allJasparDataBase <- read_excel("~/Scripts/May/allJasparDataBase.xlsx", 
     col_names = FALSE)%>%as.data.frame

allJasparDataBase <- split(allJasparDataBase, seq(nrow(allJasparDataBase)))
allJasparDataBase<-lapply(allJasparDataBase, as.character)%>%unique


##Loading in the ChIP-chip peaks

library(BSgenome.Mmusculus.UCSC.mm9)
library(readxl)

n2aChipSeq <- read_excel("~/DataFiles/ChIPseq/Mouse/ChIPseqDataQuille2011.xls", 
                         sheet = "only N2a")%>%as.data.frame
embyroChipSeq  <- read_excel("~/DataFiles/ChIPseq/Mouse/ChIPseqDataQuille2011.xls", sheet = "only emb brain")%>%as.data.frame
commonChipSeq <- read_excel("~/DataFiles/ChIPseq/Mouse/ChIPseqDataQuille2011.xls",  sheet = "common genes")%>%as.data.frame



chipSeqDataCleaner<-function(x){
  splitColoumnMinus<-data.frame(do.call('rbind', strsplit(as.character(x$location),'-',fixed=TRUE)))
  colnames(splitColoumnMinus)<- c("X1", "end")#re naming the coloumns
  splitColoumnSemiColon<-data.frame(do.call('rbind', strsplit(as.character(splitColoumnMinus$X1),':',fixed=TRUE)))
  colnames(splitColoumnSemiColon)<- c("chromosome", "start")#renaming those two
  geneSymbolMetaDataFromOriginalData<-x[2:3]
  dataFrameOfChipSeqData<- cbind(geneSymbolMetaDataFromOriginalData, splitColoumnSemiColon, splitColoumnMinus[2])%>%na.omit()
  removingTheNegatives<- cbind(dataFrameOfChipSeqData, (as.data.frame(as.numeric(as.character(dataFrameOfChipSeqData$end)))-as.data.frame(as.numeric(as.character(dataFrameOfChipSeqData$start)))))
  
  
  negativesRemoved<-subset(removingTheNegatives, removingTheNegatives$`as.numeric(as.character(dataFrameOfChipSeqData$end))`>0)
  grangeChipSeq<-makeGRangesFromDataFrame(negativesRemoved,
                                          keep.extra.columns=FALSE,
                                          ignore.strand=FALSE,
                                          seqinfo=NULL,
                                          seqnames.field=c("seqnames", "seqname",
                                                           "chromosome", "chrom",
                                                           "chr", "chromosome_name",
                                                           "seqid"),
                                          start.field="start",
                                          end.field=c("end", "stop"),
                                          strand.field="strand",
                                          starts.in.df.are.0based=FALSE)
}



grangeN2aChipSeq<-chipSeqDataCleaner(n2aChipSeq)
grangeBrainChipSeq<-chipSeqDataCleaner(embyroChipSeq)
grangeCommonChipSeq<-chipSeqDataCleaner(commonChipSeq)


## getting the PWM

NumberOfPeaksID<-function(x){
  
abdB<-list()
abdB[["ID"]]<- x
abdBMatrix<- getMatrixByID(JASPAR2016, abdB)
abdBMatrix<- abdBMatrix@profileMatrix


## get PWM

object<-matchPWM(abdBMatrix, BSgenome.Mmusculus.UCSC.mm9, "100%")


return(  cbind(
n2aAzf1Overlaps<-length(subset(grangeN2aChipSeq, findOverlaps(object, grangeN2aChipSeq)%>%countRnodeHits())),
brainAzf1Overlaps<-subset(grangeBrainChipSeq, findOverlaps(object, grangeBrainChipSeq)%>% countRnodeHits())%>%length(),
commonAzf1Overlaps<-subset(grangeCommonChipSeq, findOverlaps(object, grangeBrainChipSeq)%>%countRnodeHits())%>%length()
)
)
}



## Doing it as a list
permantDataFrame=NULL
i=1
for(i in 1:length(allJasparDataBase)) {
  
abdB<-list()
abdB[["ID"]]<- allJasparDataBase[[i]]
abdBMatrix<- getMatrixByID(JASPAR2016, abdB)
abdBMatrix<- abdBMatrix@profileMatrix


## get PWM

object<-matchPWM(abdBMatrix, BSgenome.Mmusculus.UCSC.mm9, "100%")


tempdataFrame<-cbind(
n2aAzf1Overlaps<-length(subset(grangeN2aChipSeq, findOverlaps(object, grangeN2aChipSeq)%>%countRnodeHits())),
brainAzf1Overlaps<-subset(grangeBrainChipSeq, findOverlaps(object, grangeBrainChipSeq)%>% countRnodeHits())%>%length(),
commonAzf1Overlaps<-subset(grangeCommonChipSeq, findOverlaps(object, grangeBrainChipSeq)%>%countRnodeHits())%>%length()
)
permantDataFrame<-rbind(permantDataFrame, tempdataFrame)
i<-i+1
}
colnames(permantDataFrame)<-allJasparDataBase

library(pander)
permantDataFrame%>%pander()
```


